common/fixation_entry.go

```diff
@@ -269,10 +330,9 @@ func (fs *FixationStore) getUnmarshaledEntryForBlock(ctx sdk.Context, safeIndex
 		fs.cdc.MustUnmarshal(iterator.Value(), &entry)
 
 		if entry.Block <= block {
-			if entry.IsStale(ctx) {
+			if entry.IsStale(ctx) && !entry.Deleted {
```

Comment:
can you explain why?
this means : dont show the entry if it is stale and not deleted
why we want to show it if it is stale and deleted?

Reply:
because AppendEntry() expects to always dig the latest entry. and if that latest entry is stale (because it's was deleted enough blocks ago), then it may get a non-latest entry and its logic will break (e.g. it assume not-found means no entry versions for that index exists, which may not be the case).